staker/validator_instance.go

```diff
@@ -0,0 +1,48 @@
+package staker
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/offchainlabs/nitro/validator"
+	"github.com/offchainlabs/nitro/validator/client/redis"
+	"github.com/offchainlabs/nitro/validator/inputs"
+)
+
+type LegacyValidatorInstance interface {
+	CheckLegacyValid() error
+	LegacyLastValidatedInfo() (*legacyLastBlockValidatedDbInfo, error)
+	SetLegacyValidInfo(info *legacyLastBlockValidatedDbInfo)
+}
+
+// ValidatorInstance is an interface used by BlockValidator to carry out block and MEL-state validations. BlockValidatorInstance implements
+// this interface to carry out block validations and MELValidatorInstance (TODO) implements this to carry out MEL-state validation
+type ValidatorInstance interface {
+	// Methods corresponding to ValidatorInstance's overall context
+	IsValidatedGlobalStateNew(gs validator.GoGlobalState) bool
+	LastValidatedGlobalState() validator.GoGlobalState
+	SetLastValidatedGlobalState(gs validator.GoGlobalState)
+	LastValidatedCount() (count uint64, isChainCaughtupToLastValidateCount bool, err error)
```

Comment:
typo: change isChainCaughtupToLastValidateCount to isChainCaughtupToLastValidatedCount